keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java

```diff
@@ -101,46 +104,222 @@ private void closeClients() {
+    assertThat(secretEntry).isPresent();
+    assertThat(secretEntry.get().getValue()).isEqualTo(SECRET_VALUE);
+  }
+
   @BeforeAll
   void setup() {
     verifyEnvironmentVariables();
     setupSecretsManagers();
     setupSecretsManagerClient();
-    createSecret();
+    initializeVariables();
   }
 
   @AfterAll
   void teardown() {
     if (awsSecretsManagerDefault != null
         || awsSecretsManagerExplicit != null
         || secretsManagerClient != null) {
-      deleteSecret();
+      deleteSecrets();
       closeClients();
     }
   }
 
   @Test
   void fetchSecretWithDefaultManager() {
+    createSecret(false, false, false);
     Optional<String> secret = awsSecretsManagerDefault.fetchSecret(secretName);
     assertThat(secret).hasValue(SECRET_VALUE);
   }
 
   @Test
   void fetchSecretWithExplicitManager() {
+    createSecret(false, false, false);
     Optional<String> secret = awsSecretsManagerExplicit.fetchSecret(secretName);
     assertThat(secret).hasValue(SECRET_VALUE);
   }
 
   @Test
   void fetchSecretWithInvalidCredentialsReturnsEmpty() {
+    createSecret(false, false, false);
     assertThatExceptionOfType(RuntimeException.class)
         .isThrownBy(() -> awsSecretsManagerInvalidCredentials.fetchSecret(secretName))
         .withMessageContaining("Failed to fetch secret from AWS Secrets Manager.");
   }
 
   @Test
   void fetchingNonExistentSecretReturnsEmpty() {
+    createSecret(false, false, false);
     Optional<String> secret = awsSecretsManagerDefault.fetchSecret("signers-aws-integration/empty");
     assertThat(secret).isEmpty();
   }
+
+  @Test
+  void listAndMapSingleSecretWithSingleTag() {
+    createSecret(false, false, false);
+
+    final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries =
+        awsSecretsManagerExplicit.mapSecrets(
+            secretTags.keySet().stream().collect(Collectors.toList()),
+            secretTags.values().stream().collect(Collectors.toList()),
+            AbstractMap.SimpleEntry::new);
+
+    secretNames.forEach(secretName -> validateMappedSecret(secretEntries, secretName));
+  }
+
+  @Test
+  void listAndMapSingleSecretWithMultipleTags() {
+    createSecret(false, true, false);
+
+    final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries =
+        awsSecretsManagerExplicit.mapSecrets(
+            secretTags.keySet().stream().collect(Collectors.toList()),
+            secretTags.values().stream().collect(Collectors.toList()),
+            AbstractMap.SimpleEntry::new);
+
+    secretNames.forEach(secretName -> validateMappedSecret(secretEntries, secretName));
+  }
+
+  @Test
+  void listAndMapMultipleSecretsWithMultipleTags() {
+    createSecret(true, true, false);
+
+    final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries =
+        awsSecretsManagerExplicit.mapSecrets(
+            secretTags.keySet().stream().collect(Collectors.toList()),
+            secretTags.values().stream().collect(Collectors.toList()),
```

Comment:
nit: similarly this can be simpler now that collection is being used, can use secretTags.values() and same for the other tests